### PR TITLE
Add jp.yvt.OpenSpades

### PR DESCRIPTION
**Pull request by digiberk**
OpenSpades is an open-source clone of Ace of Spades

I am not the original developer of this project but because no one has added it to Flathub, I thought I would go ahead and do it. The flatpak manifest file(s) are in the openspades branch. If anything is missing, some other files are here.

Credit goes to @kdeleteme and @theunknownxy for creating the manifest file. I have made a few changes to fix compatibility with different versions of ImageMagick. (https://github.com/yvt/openspades/blob/master/flatpak/jp.yvt.OpenSpades.json)

(This is the main PR for OpenSpades)

### CODE REVIEW

**Comment by digiberk** [comment]
bot, build jp.yvt.OpenSpades

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "skip-arches": ["arm"]
```
Comment (theunknownxy) [inline flathub.json]: Oh I should have mentioned: `aarch64` is basically 64-bit ARM
```suggestion
  "skip-arches": ["arm", "aarch64"]
```

Comment (digiberk) [reply flathub.json]: Thanks! That should fix it now.

**Comment by digiberk** [comment]
bot, build jp.yvt.OpenSpades

**Comment by digiberk** [comment]
Okay, so I have tested it and everything runs as expected, there seems to be no issues.

File: jp.yvt.OpenSpades.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,78 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>jp.yvt.OpenSpades.desktop</id>
```
Comment (nedrichards) [inline jp.yvt.OpenSpades.appdata.xml]: is this the actual app-id in the upstream code? This is *not* the recommended format for a flatpak app-id https://github.com/flathub/flathub/wiki/AppData-Guidelines

**Comment by nedrichards** [comment]
Is there an issue upstream to add the appdata? That's useful for everything on desktop Linux and upstream's really should be providing it rather than hosting it as a downstream patch on flathub.

**Comment by theunknownxy** [comment]
> Is there an issue upstream to add the appdata? That's useful for everything on desktop Linux and upstream's really should be providing it rather than hosting it as a downstream patch on flathub.

The appdata is sort of part of the repository, but it's only used for the flatpak packaging (which is basically all files from this PR in a subdirectory). But yeah, the flatpak build script here could also just use those.

Generally speaking the current build scripts of OpenSpades are more tailored for running OpenSpades right from the build folder or installing it only on the local machine, and not for packaging and distributing (For example the NonFree package is automatically downloaded and installed).

File: jp.yvt.OpenSpades.appdata.xml (hunk 2 above)
Comment (theunknownxy) [reply jp.yvt.OpenSpades.appdata.xml]: Basically this?
```suggestion
  <id>jp.yvt.OpenSpades</id>
```

Comment (nedrichards) [reply jp.yvt.OpenSpades.appdata.xml]: Yep!

**Comment by nedrichards** [comment]
@digiberk when you push a change for the app-id I'll rebuild, test manually and should be good to go.

**Comment by digiberk** [comment]
> @digiberk when you push a change for the app-id I'll rebuild, test manually and should be good to go.

When you mean app-id @nedrichards ? I have the app-id in the repository already in the correct manifest file (jp.yvt.OpenSpades.json). Am I missing something here? I'm still quite new to Flatpak packaging so any help would be appreciated. I have looked into the requirements and manifests documentation but I am unsure why I should push a change for the app-id, it already is there. 

Also, I have tested last build and it works without any problems. :smile:

**Comment by digiberk** [comment]
bot, build jp.yvt.OpenSpades

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/jp.yvt.OpenSpades

**Pull request closed without merging**